### PR TITLE
Ice Cream (Color Gradients)

### PR DESCRIPTION
**Pull request by XSlayer300**
## About The Pull Request
Adds gradient hair as an option in the preferences set up.

## Why It's Good For The Game
Because snowflake ahoy, ported from Polaris/VOREStation.

## Changelog
:cl:
add: TerraGov made one-step further to lessen the restrictions on TerraGov's military hairstyle (and it had gone too far), hair color gradients are now available in the preferences menu!
/:cl:

### CODE REVIEW

**Comment by tgstation-server** [comment]
You currently have a negative Fix/Feature pull request delta of -27. Maintainers may close this PR at will. Fixing issues or improving the codebase will improve this score.

**Review by SplinterGP — CHANGES_REQUESTED** [review]

**Comment by Kuro-ai** [comment]
God help us all

**Comment by XSlayer300** [comment]
> God help us all

exactly.

**Comment by BeloneX** [comment]
Can you precisely explain what this does?

**Comment by SplinterGP** [comment]
> Can you precisely explain what this does?

gradient patterns for hair

File: code/modules/mob/living/carbon/human/update_icons.dm [hunk 1]
```diff
@@ -332,7 +332,15 @@ GLOBAL_LIST_EMPTY(damage_icon_parts)
 		if(hair_style && (species.name in hair_style.species_allowed))
 			var/icon/hair_s = new/icon("icon" = hair_style.icon, "icon_state" = "[hair_style.icon_state]_s")
 			if(hair_style.do_colouration)
+				var/icon/grad_s
+				if(grad_style && grad_style != "None")
+					var/datum/sprite_accessory/gradient = GLOB.hair_gradients_list[grad_style]
+					grad_s = new/icon("icon" = 'icons/mob/hair_gradients.dmi', "icon_state" = gradient.icon_state)
```
Comment (SplinterGP) [inline code/modules/mob/living/carbon/human/update_icons.dm:338]:
```suggestion
					grad_s = new/icon("icon" = gradient.icon, "icon_state" = gradient.icon_state)
```

File: code/__HELPERS/global_lists.dm [hunk 2]
```diff
@@ -19,6 +19,12 @@ GLOBAL_LIST_EMPTY(randomized_pill_icons)
 	for(var/path in subtypesof(/datum/sprite_accessory/hair))
 		var/datum/sprite_accessory/hair/H = new path()
 		GLOB.hair_styles_list[H.name] = H
+
+	// Hair Gradients - Initialise all /datum/sprite_accessory/hair_gradient into an list indexed by hair-style name
```
Comment (SplinterGP) [inline code/__HELPERS/global_lists.dm:23]:
```suggestion
	// Hair Gradients - Initialise all /datum/sprite_accessory/hair_gradient into an list indexed by gradient-style name
```

**Review by SplinterGP — CHANGES_REQUESTED** [review]

**Review by SplinterGP — APPROVED** [review]

**Comment by tgstation-server** [comment]
Your Fix/Feature pull request delta is currently below zero (-36). Maintainers may close future Feature/Tweak/Balance PRs. Fixing issues or helping to improve the codebase will raise this score.

**Pull request merged**